ql/ql/src/codeql_ql/StructuredLogs.qll

```diff
@@ -0,0 +1,138 @@
+private import ql
+private import codeql_ql.ast.internal.TreeSitter
+
+class Object extends JSON::Object {
+  JSON::Value getValue(string key) {
+    exists(JSON::Pair p |
+      p = this.getChild(_) and p.getKey().(JSON::String).getChild().getValue() = key
+    |
+      result = p.getValue()
+    )
+  }
+
+  string getString(string key) { result = this.getValue(key).(JSON::String).getChild().getValue() }
+
+  int getNumber(string key) { result = this.getValue(key).(JSON::Number).getValue().toInt() }
+
+  Array getArray(string key) { result = this.getValue(key) }
+
+  Object getObject(string key) { result = this.getValue(key) }
+
+  string getType() { result = this.getString("type") }
+
+  int getEventId() { result = this.getNumber("event_id") }
+
+  string getTime() { result = this.getString("time") }
+}
+
+class Array extends JSON::Array {
+  Object getObject(int i) { result = this.getChild(i) }
+
+  string getString(int i) { result = this.getChild(i).(JSON::String).getChild().getValue() }
+
+  int getNumber(int i) { result = this.getChild(i).(JSON::Number).getValue().toInt() }
+
+  Array getArray(int i) { result = this.getChild(i) }
+}
+
+abstract class LogEntry extends Object { }
```

Comment:
Two things:
1. What's the QL-for-QL policy on abstract AST classes?
2. Is "LogEntry" a too general name? There are other summaries that we definitely want to ingest as well as part of this work (i.e., the ones generated by `codeql generate log-summary` that also generate jsonl files with log entries).

Reply:
Just to answer these questions (even though the PR was merged already)
1. I believe the general consensus on abstract classes is "don't". In this case, I just wanted to have an easy way to union together the various subclasses to see if I'd covered anything. We can get rid of the abstract class now, I think.
2. We should probably push this into an inner module to make things a bit more clean. That way, the generality of the name isn't a big problem either.